core/field.js

```diff
@@ -186,11 +221,21 @@ Blockly.Field.prototype.clickTarget_ = null;
  * A developer hook to override the returned text of this field.
  * Override if the text representation of the value of this field
  * is not just a string cast of its value.
+ * Return null to resort to a string cast.
  * @return {?string} Current text. Return null to resort to a string cast.
  * @protected
  */
 Blockly.Field.prototype.getText_;
 
+/**
+ * An optional method that can be defined to show an editor when the field is
+ *     clicked. Blockly will automatically set the field as clickable if this
+ *     method is defined.
+ * @return {void}
+ * @protected
+ */
+Blockly.Field.prototype.showEditor_;
```

Comment:
Could you check that `isClickable` works w/ this change? And verify that the field clicking works? I think `isClickable` can maybe be simplified now.

Reply:
When it's defined like this, showEditor_ is undefined in JS, and so it doesn't affect any of the logic in isClickable.
This is purely just a hint to the closure compiler.

Reply:
Oh sweet! Good to know.